### PR TITLE
Week 2 lex

### PR DESCRIPTION
**Pull request by LPeterson18**
All Questions Complete

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,6 +15,15 @@
 
 // --------------------1) Create a function that takes a number as an argument and decides if the number is evenly divisble by three or not.
 
+//Pseudocode
```
Comment (minachuong) [inline code-challenges.test.js:18]: awesome pseudocode!

File: code-challenges.test.js [hunk 2]
```diff
@@ -24,20 +33,57 @@ var num2 = 0
...
 
 var num3 = -7
-// Expected output: "-7 is not divisble by three"
-
-
-
-
+// Expected output: "-7 is not divisible by three"
+
+describe("When divisibleByThree is called", () => {
```
Comment (minachuong) [inline code-challenges.test.js:38]: I like that you literally wrote the test above the function! :D

File: code-challenges.test.js [hunk 3]
```diff
@@ -24,20 +33,57 @@ var num2 = 0
...
+  it("return `15 is divisible by three` when 15 called is divisible by three",() => {
```
Comment (minachuong) [inline code-challenges.test.js:39]: very descriptive tests! 100

File: code-challenges.test.js [hunk 4]
```diff
@@ -24,20 +33,57 @@ var num2 = 0
...
+    return `${number} is divisible by three`
+  }
+}
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
+//Pseudocode
+//create a function
+//argument is an array of words
+//.map of array to alter every element in array
```
Comment (minachuong) [inline code-challenges.test.js:80]: 💯

File: code-challenges.test.js [hunk 5]
```diff
@@ -46,38 +92,101 @@ var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
+    //assert
+  expect(actualResult).toEqual(["Temperature", "Database", "Chopsticks", "Mango", "Deduction"])
+  })
+})
 
 
 // b) Create the function that makes the test pass.
 
-
+const capitalizeArray = (array) => {
```
Comment (minachuong) [inline code-challenges.test.js:117]: 👍 on function naming

File: code-challenges.test.js [hunk 6]
```diff
@@ -46,38 +92,101 @@ var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
 
 
 // b) Create the function that makes the test pass.
 
-
+const numberSorter = (array) => {
+  return array.filter(value => {
+    return typeof value === "number"
+  }).sort((a, b) => a-b)
+//In order to sort correctly have to use the function keyword to define a function within .sort.  This allows for
```
Comment (minachuong) [inline code-challenges.test.js:174]: looks like this is correct without needing to use `function` keyword because `function() {}` and `() => {}` in this context will behave the same way. But in case you're interested in doing more research `function() {}` is considered an anonymous function expression and `() => {}` is an arrow function

File: code-challenges.test.js [hunk 7]
```diff
@@ -87,8 +196,45 @@ var vowelTester2 = "academy"
...
+  })
+})
 
 // b) Create the function that makes the test pass.
+
+const vowelIndexFinder = (string) => {
+  var i = 0;
+  var vowelIndex
+  var letterArray = string.split("")
+  while (i < letterArray.length) {
```
Comment (minachuong) [inline code-challenges.test.js:232]: interesting use of while-loop! This is a great optimization because the code will stop iterating through the array as soon as it encounters the first index. You may be able to simplify the code by just returning `i` without needing to use `vowelIndex` since they represent the same thing. also there's a keyword `break` that can be used in loops that short circuits the iteration

File: interview-questions.md [hunk 8]
```diff
@@ -9,57 +9,77 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. Why would you use the method super()?
 
   Your answer:
+  super() is necessary in react when calling on classes.  It allows old constructors to be inherited.
```
Comment (minachuong) [inline interview-questions.md:12]: not just in react but in plain Javascript too. It's not necessarily that it inherits the constructor but rather inherits properties and behavior

File: react-challenge.html [hunk 9]
```diff
@@ -35,24 +35,27 @@
 
         handleMoodOptions = () => {
           let randomIndex = Math.floor(Math.random() * this.state.moods.length )
-          this.setState({ currentMood: moods[randomIndex] })
+          this.setState({ currentMood: this.state.moods[randomIndex] })
+          //changed moods[randomIndex] to this.state.moods[randomIndex]
```
Comment (minachuong) [inline react-challenge.html:39]: awesome pseudocode!

**Review by minachuong — APPROVED** [review]
awesome pseudocoding and exercising best practices. Also very cool optimization on minimizing unnecessary iteration. If you're looking for a challenge, consider safe guarding your function logic by using tests to discover those edge cases. Great job.
